File: Algorithm/Greedy/체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var students = Array(repeating: 1, count: n + 1)
    students[0] = 0
    
    for l in lost {
        students[l] -= 1
    }
    
    for r in reserve {
        students[r] += 1
    }
    
    for i in 1...n {
        if students[i] != 0 {
            continue
        }
        
        if (i > 1 && students[i - 1] > 1) {
            students[i - 1] -= 1
            students[i] = 1
        } else if (i < n && students[i + 1] > 1) {
            students[i + 1] -= 1
            students[i] = 1
        }
    }
    
    return students.filter{$0 > 0}.count
}

File: Algorithm/Greedy/체육복.py
def solution(n, lost, reserve):
    answer = 0
    arr = [1]*n
    
    for l in lost:
        arr[l-1] -= 1
    for r in reserve:
        arr[r-1] += 1
    
    for i in range(len(arr)):
        if arr[i] == 0:
            if i > 0 and arr[i-1] >= 2:
                arr[i-1] -= 1
                arr[i] += 1
                
            elif i < n-1 and arr[i+1] >= 2:
                    arr[i+1] -= 1
                    arr[i] += 1
    
    for a in arr:
        if a > 0:
            answer += 1
    return answer